internal/subscriber/subscriber.go

```diff
@@ -87,78 +76,48 @@ func (s *Subscriber) GetID() string {
 
 // GetSubscription ...
 func (s *Subscriber) GetSubscription() string {
-	return s.subscription
+	return s.subscription.Name
 }
 
-// commits existing message on primary topic and pushes message to the pre-defined retry topic
-func (s *Subscriber) retry(ctx context.Context, retryMsg *RetryMessage) {
-	if retryMsg == nil {
+// pushes message to retrier and commit existing message on primary topic
+func (s *Subscriber) retry(ctx context.Context, msg messagebroker.ReceivedMessage) {
+
+	// for older subscriptions, delayConfig will not get auto-created
+	if s.retrier == nil {
+		logger.Ctx(ctx).Infow("subscriber: skipping retry as retrier not configured", "logFields", s.getLogFields())
 		return
 	}
 
-	logFields := s.getLogFields()
-	logFields["messageId"] = retryMsg.MessageID
-
-	startTime := time.Now()
-
-	// remove message from the primary topic
-	_, err := s.consumer.CommitByPartitionAndOffset(ctx, messagebroker.CommitOnTopicRequest{
-		Topic:     retryMsg.Topic,
-		Partition: retryMsg.Partition,
-		// add 1 to current offset
-		// https://docs.confluent.io/5.5.0/clients/confluent-kafka-go/index.html#pkg-overview
-		Offset: retryMsg.Offset + 1,
-	})
-
+	// prepare message headers to be used by retrier
```

Comment:
can you add comments on what purpose each header is used for. looking at the current list not able to understand why we need so much data in headers, can just keeping subscription name do the trick ?

Reply:
added.